lib/floe/workflow.rb

```diff
@@ -34,43 +51,45 @@ def initialize(payload, context = nil, credentials = {})
       raise Floe::InvalidWorkflowError, err.message
     end
 
-    def step
-      context.execution["StartTime"] ||= Time.now.utc
-
-      context.state["Guid"]    = SecureRandom.uuid
-      context.state["Input"] ||= context.execution["Input"].dup
+    def run!
+      until end?
+        step
+      end
```

Comment:
`step` is non-blocking.

So I think this could spin a lot of cpu.

Probably for another spike PR:
Does it make sense to return the number of seconds to check back in?
If it encoutered a wait/retry, then it could return the value
If it is a task (and unknown), then it just returns 1?

I say this is for another PR because that would potentially make it a more involved/confusing interface.